Feature/search sample characteristics

henrikjohansson712 opened the pull request:
## Description

Add filtering on sample characteristics to sample table.

Requires [catamel #349](https://github.com/SciCatProject/catamel/pull/349)

## Motivation

#475 

## Changes:

* Create new shared Search Parameters Dialog module
* Add button for characteristics filtering to sample dashboard
* Add chip list for applied characteristics filters to sample dashboard
* Update sample sdk with new metadataKeys endpoint
* Rename "Add Sample" button to "Create Sample", for consistency with dataset equivalent and to avoid confusion with new "Add Characteristic" filter
* Create GenericFilter interface containing filters that are common among several models

## Tests included/Docs Updated?

- [x] Included for each change/fix?
- [x] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?
- [x] Requires update of catamel API?

stephan271 commented:
Is there a flag in environments related to this feature , which I need to switch on for tests ? Do we need to care for the "coverage" CI/CD complaints ?

henrikjohansson712 commented:
Added the same flag as for the dataset version of the search: `scienceSearchEnabled`. The coverage always fluctuates a bit, but I realised that I forgot the unit tests for the new module, so I added those.
